src/clean-core/assert.hh

```diff
@@ -11,17 +11,20 @@
 // CC_ENABLE_BOUND_CHECKING enables bound checking
 
 
-// the debugger should break on-site, so this cannot hide in a function call
-// comma expression: abort must still be called (if no debugger is attached), and do/while(0) doesn't work in the ternary
+// the debugger should break right in the assert macro, so this cannot hide in a function call
 
 #ifdef CC_COMPILER_MSVC
-#define CC_BREAK_AND_ABORT() (__debugbreak(), ::cc::detail::perform_abort())
+// __debugbreak() terminates immediately without an attached debugger
+#define CC_BREAK() (::cc::detail::is_debugger_connected() ? __debugbreak() : void(0))
```

Comment:
`CC_BREAK` sounds like it could also be related to normal control flow. it is rarely used directly by the user so we could go for the longer `CC_DEBUGGER_BREAK` or similar.

Reply:
Renamed